File: mall-content-interface/src/main/java/com/mall/content/service/ContentCategoryService.java
package com.mall.content.service;

import com.mall.pojo.ContentCategory;
import common.pojo.PageResult;

import java.util.List;

/**
 * 内容/广告类目服务接口
 * @author Wwl
 *
 */
public interface ContentCategoryService {

	/**
	 * 增加
	 * @param contentCategory
	 */
	public void add(ContentCategory contentCategory);

	/**
	 * 修改
	 * @param contentCategory
	 */
	public void update(ContentCategory contentCategory);

	/**
	 * 根据Id获取实体
	 * @param id
	 * */
	public ContentCategory findOne(Long id);

	/**
	 * 查询所有
	 * @return
	 */
	public List<ContentCategory> findAll();

	/**
	 * 查询所有-分页
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pageNum, int pageSize);

	/**
	 * 根据条件查询，分页
	 * @param contentCategory
	 * @param pageNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(ContentCategory contentCategory, int pageNum, int pageSize);

	/**
	 * 删除
	 * @return
	 */
	public void delete(Long[] ids);

	
}

File: mall-manager-web/src/main/webapp/js/service/contentCategoryService.js
/**
 * 内容/广告类目服务层
 * Created by Wwl on 2019/3/4
 */
app.service('contentCategoryService', function($http){

    //增加
    this.add = (entity) => $http.post('../contentCategory/add.do',entity);

    //修改
    this.update = (entity) => $http.post('../contentCategory/update.do',entity);

    //删除
    this.delete = (ids) => $http.get('../contentCategory/delete.do?ids='+ids);

    //查询实体
    this.findOne = (id) => $http.get('../contentCategory/findOne.do?id='+id);

    //查询列表
    this.findList = () => $http.get('../contentCategory/list.do');

    //分页查询
    this.findPage = (page,size) => $http.get('../contentCategory/listByPage.do?page=' + page + '&size=' + size);

    //搜索
    this.search = (page,size,searchEntity) => $http.post('../contentCategory/search.do?page='+page+'&size='+size,searchEntity);

    //下拉列表数据
    this.findOptionList = () => $http.get('../contentCategory/findOptionList.do');

});